src/popup/components/TransactionOverview.vue

```diff
@@ -8,103 +8,161 @@
   />
 </template>
 
-<script>
-import { mapState, mapGetters } from 'vuex';
+<script lang="ts">
 import { SCHEMA } from '@aeternity/aepp-sdk';
-import { watchUntilTruthy } from '../utils';
+import {
+  computed,
+  defineComponent,
+  onMounted,
+  ref,
+} from '@vue/composition-api';
+import { TranslateResult } from 'vue-i18n';
+import {
+  TRANSACTION_DIRECTION_RECEIVED,
+  TRANSACTION_DIRECTION_SENT,
+  TX_FUNCTION_CLAIM,
+  watchUntilTruthy,
+} from '../utils';
+import { useSdk, useTransaction } from '../../composables';
+import { useState, useGetter } from '../../composables/vuex';
+import {
+  IAccount, IAccountLabeled, ITransaction, TransactionType,
+} from '../../types';
 import TransactionInfo from './TransactionInfo.vue';
 
-export default {
+export default defineComponent({
   components: { TransactionInfo },
   props: {
     tx: { type: Object, required: true },
-    isDex: { type: Boolean, default: false },
   },
-  data: () => ({ name: '' }),
-  computed: {
-    ...mapGetters(['getTxType', 'getTxDirection', 'getExplorerPath', 'getDexContracts']),
-    ...mapGetters('names', ['getPreferred']),
-    ...mapState({
-      account(_, { account }) {
-        return {
-          ...account,
-          label: this.$t('transaction.overview.accountAddress'),
-          url: this.getExplorerPath(account.address),
-        };
-      },
-      isDexRecipient() {
-        return [...this.getDexContracts.router, ...this.getDexContracts.wae]
-          .includes(this.tx?.contractId);
-      },
-    }),
-    transaction() {
-      switch (this.txType) {
+  setup(props, { root }) {
+    const name = ref('');
+    const ownershipAccount = ref<IAccountLabeled | IAccount | {}>({});
```

Comment:
what do you think about adding type in types.ts : 
type EmptyObject = {}

Reply:
I think that won't give us anything